### PR TITLE
Refactor access map and time filtering in DbService and permission service

### PR DESCRIPTION
**Pull request by ivanslabbert**

### CODE REVIEW

File: api/src/db/db.service.ts [hunk 1]
```diff
@@ -323,27 +324,43 @@ export class DbService extends EventEmitter {
...
+                });
+            }
+
+            if (options.to) {
+                selectors.push({
+                    updatedTimeUtc: {
+                        $lt: options.to + this.syncTolerance,
+                    },
+                });
+            }
```
Comment (laurensgroeneveld) [inline api/src/db/db.service.ts:356]: I'm a bit confused by this. `selectors` does not seem to be used, does it work by reference? Why is it needed to have a separate variable for it?

File: api/src/socketio.ts
Comment (laurensgroeneveld) [inline api/src/socketio.ts:1]: I'm missing tests for the changes in this file.

**Review by laurensgroeneveld — CHANGES_REQUESTED** [review]
Missing some tests and a small question.

File: api/src/db/db.service.ts (hunk 1 above)
Comment (ivanslabbert) [reply api/src/db/db.service.ts:356]: Simplified this a bit to make it more readable

File: api/src/socketio.ts
Comment (ivanslabbert) [reply api/src/socketio.ts:1]: - Added a test to check if an AccessMap is returned after clientDataReq.
- Added a skipped test as placeholder for test to check if historical data is returned after a user is assigned additional groups. This should be implemented after JWT verification (and group assignments) are implemented in the API. As all requests are hard-coded with "group-super-admins" permissions at the moment, testing this is a bit harder, and will have to be changed as soon as JWT verification is implemented.

Comment (laurensgroeneveld) [reply api/src/socketio.ts:1]: > This should be implemented after JWT verification

Can you add it as a note to the authentication ticket? So we don't forget about it.

**Review by laurensgroeneveld — APPROVED** [review]
LGTM

**Pull request merged**